### ISSUE TITLE
yhat.default applies as.numeric to predicted values, which may cause errors

### ISSUE THREAD

**Issue by hbaniecki** [ModelOriented/DALEX#149]
https://github.com/ModelOriented/DALEX/blob/27b401d0a161457f09ec9e284aecf3ddf61ac3f5/R/yhat.R#L157-L160

This function transforms a proper 3-column data frame into a 1-dimensional vector (3 times longer), which makes no sense. It may cause errors and could work better (possibly some if statements).
Example: 
```r
library(iBreakDown)
library(DALEX)

model_rpart <- rpart::rpart(Species~., data = iris)

ex <- explain(model, data = iris, y = iris$Species)

predict(model_rpart, iris) # makes sense
DALEX::yhat(model_rpart, iris) # invalid

bd <- break_down(ex, new_observation = iris[1,])
plot(bd) # invalid because of yhat.default

#:# working example
ex2 <- explain(model, data = iris, y = iris$Species,
              predict_function = rpart:::predict.rpart)
bd2 <- break_down(ex2, new_observation = iris[1,])
plot(bd2)
```

**Comment by pbiecek** [comment]
@maksymiuks can you tak care about this one?
we need
1. yhat for rpart
2. default yhat should keep dimensions, if predict returns a vector of a single column, then this shall be converted into a vector. but for multiple columns (multilabel classification) default yhat shall return all collumns

**Comment by maksymiuks** [comment]
Candidate fix in https://github.com/ModelOriented/DALEX/commit/d051288690d84aa2628f12ce36e227655f14180d

**Comment by pbiecek** [comment]
fixed in 78eaeca
